src/main/java/org/spongepowered/api/service/ServiceProvider.java

```diff
@@ -0,0 +1,86 @@
+package org.spongepowered.api.service;
+
+import org.spongepowered.api.service.ban.BanService;
+import org.spongepowered.api.service.economy.EconomyService;
+import org.spongepowered.api.service.pagination.PaginationService;
+import org.spongepowered.api.service.permission.PermissionService;
+import org.spongepowered.api.service.rcon.RconService;
+import org.spongepowered.api.service.sql.SqlService;
+import org.spongepowered.api.service.user.UserStorageService;
+import org.spongepowered.api.service.whitelist.WhitelistService;
+
+import java.util.Optional;
+
+/**
+ * Provides various Sponge services.
+ */
+public interface ServiceProvider {
+
+    /**
+     * Provides the service represented by the supplied {@link Class}.
+     *
+     * @param serviceClass The class
+     * @param <T> The type of service
+     * @return The service, if one exists
+     */
+    <T> Optional<T> provide(Class<T> serviceClass);
```

Comment:
Please add a `provideUnchecked`, like before.

Reply:
No. 

As this is a PoC I've not fully explained what's going on here but this is not intended for third party service interfaces. The only reason I've added this is to enable third party implementations to add extra first-party service interfaces.